src/core/taint.c

```diff
@@ -0,0 +1,87 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include <sys/utsname.h>
+
+#include "alloc-util.h"
+#include "clock-util.h"
+#include "cgroup-util.h"
+#include "errno-util.h"
+#include "fileio.h"
+#include "fs-util.h"
+#include "log.h"
+#include "os-util.h"
+#include "path-util.h"
+#include "strv.h"
+#include "taint.h"
+#include "uid-range.h"
+
+static int short_uid_range(const char *path) {
+        _cleanup_(uid_range_freep) UIDRange *p = NULL;
+        int r;
+
+        assert(path);
+
+        /* Taint systemd if we the UID range assigned to this environment doesn't at least cover 0…65534,
+         * i.e. from root to nobody. */
+
+        r = uid_range_load_userns(path, UID_RANGE_USERNS_INSIDE, &p);
```

Comment:
Hmm, if you touch this, I'd like to make `short_uid_range` take UID/GID_RANGE_USERNS_INSIDE, rather than /proc/self/*_map

Reply:
yeah, agree

Reply:
Updated.